kentikapi/internal/api_payloads/saved_filter.go

```diff
@@ -0,0 +1,182 @@
+	SavedFilterPayload
+}
+
+type UpdateSavedFilterRequest struct {
+	SavedFilterPayload
+}
+
+type UpdateSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type SavedFilterPayload struct {
+	ID                IntAsString    `json:"id,omitempty" response:"get,post,put"`
+	CompanyID         models.ID      `json:"company_id,string,omitempty" response:"get,post,put"`
+	FilterName        string         `json:"filter_name" response:"get,post,put"`
+	FilterDescription string         `json:"filter_description" response:"get,post,put"`
+	FilterLevel       string         `json:"filter_level,omitempty" response:"get,post,put"`
+	CreatedDate       *time.Time     `json:"cdate,omitempty" response:"get,post,put"`
+	UpdatedDate       *time.Time     `json:"edate,omitempty" response:"get,post,put"`
+	Filters           FiltersPayload `json:"filters" response:"get,post,put"`
+}
+
+func (p SavedFilterPayload) ToSavedFilter() (models.SavedFilter, error) {
+	filters, err := p.Filters.ToFilters()
+	if err != nil {
+		return models.SavedFilter{}, err
+	}
+
+	return models.SavedFilter{
+		ID:                int(p.ID),
+		CompanyID:         p.CompanyID,
+		FilterName:        p.FilterName,
+		FilterDescription: p.FilterDescription,
+		FilterLevel:       p.FilterLevel,
+		CreatedDate:       *p.CreatedDate,
+		UpdatedDate:       *p.UpdatedDate,
+		Filters:           filters,
+	}, nil
+}
+
+func SavedFilterToPayload(sf models.SavedFilter) SavedFilterPayload {
+	return SavedFilterPayload{
+		ID:                IntAsString(sf.ID),
+		CompanyID:         sf.CompanyID,
+		FilterName:        sf.FilterName,
+		FilterDescription: sf.FilterDescription,
+		FilterLevel:       sf.FilterLevel,
+		CreatedDate:       &sf.CreatedDate,
+		UpdatedDate:       &sf.UpdatedDate,
+		Filters:           FiltersToPayload(sf.Filters),
+	}
+}
+
+type FiltersPayload struct {
+	Connector    string                `json:"connector" requset:"post,put" response:"get,post,put"`
+	Custom       *bool                 `json:"custom,omitempty"`
+	FilterGroups []FilterGroupsPayload `json:"filterGroups" response:"get,post,put"`
+	FilterString *string               `json:"filterString,omitempty"`
+}
+
+func (p FiltersPayload) ToFilters() (models.Filters, error) {
+	var filterGroups []models.FilterGroups
+	err := utils.ConvertList(p.FilterGroups, FilterGroupsPayload.ToFilterGroups, &filterGroups)
+	if err != nil {
+		return models.Filters{}, err
+	}
+	return models.Filters{
+		Connector:    p.Connector,
+		Custom:       p.Custom,
+		FilterGroups: filterGroups,
+		FilterString: p.FilterString,
+	}, nil
+}
+
+func FiltersToPayload(f models.Filters) FiltersPayload {
+	var filterGroups []FilterGroupsPayload
+	for _, fg := range f.FilterGroups {
+		filterGroups = append(filterGroups, FilterGroupsToPayload(fg))
+	}
+
+	return FiltersPayload{
+		Connector:    f.Connector,
+		Custom:       f.Custom,
+		FilterGroups: filterGroups,
+		FilterString: f.FilterString,
+	}
+}
+
+type FilterGroupsPayload struct {
```

Comment:
I think that in this payload no request/response validation tags make sense. See explanation above.

Reply:
Done